### ISSUE TITLE
`require-hook` errors on parameterised tests

### ISSUE THREAD

**Issue by benedictleejh** [playwright-community/eslint-plugin-playwright#481]
When trying to use parameterised tests, `require-hook` will error if it is enabled. Using the following example from Playwright's documentation:

```ts
[
  { name: 'Alice', expected: 'Hello, Alice!' },
  { name: 'Bob', expected: 'Hello, Bob!' },
  { name: 'Charlie', expected: 'Hello, Charlie!' },
].forEach(({ name, expected }) => {
  // You can also do it with test.describe() or with multiple tests as long the test name is unique.
  test(`testing with ${name}`, async ({ page }) => {
    await page.goto(`https://example.com/greet?name=${name}`);
    await expect(page.getByRole('heading')).toHaveText(expected);
  });
});
```
As can be seen from the screenshot, the lint is throwing an error saying that the parameterised test should be done in a hook.
<img width="1329" height="963" alt="Image" src="https://github.com/user-attachments/assets/77c187fb-41b1-47bd-ac2b-9ca77c4f02c9" />

I have attempted to set the `allowedFunctionCalls` setting for `require-hook` to `['forEach']`, but this did not fix the error. If it is expected that `require-hook` errors on parameterised tests by default, I would appreciate guidance on how to set it up such that it does not (I would prefer to not use `eslint-disable` every time I need parameterised tests).

**Comment by morgan-coded** [comment]
Use a `for...of` loop over a `const` cases array as the current no-disable workaround. `allowedFunctionCalls` cannot whitelist `cases.forEach(...)`: `require-hook` compares the option to `getStringValue(node.callee)`, but that helper returns names only for identifiers or string-like nodes. A `.forEach` callee is a `MemberExpression`, so it resolves to an empty string. The rule's top-level check does not report `ForOfStatement`, so the loop avoids this false positive.
